pkg/types/builder.go

```diff
@@ -277,6 +277,12 @@ func (r *resource) addParameterField(f *Field, field *types.Var) {
 }
 
 func (r *resource) addObservationField(f *Field, field *types.Var) {
+	for _, obsF := range r.obsFields {
+		if obsF.Name() == field.Name() {
+			// If the field is already added, we don't add it again.
```

Comment:
Why do observation fields get added multiple times? Is it related to [this code segment](https://github.com/upbound/upjet/blob/b89baca4ae24c8fbd8eb403c353ca18916093e5e/pkg/types/builder.go#L206)? The comment above that segment states that it prevents elimination of fields in the observation type. If we are unconditionally adding every field as an observation field, can we remove that segment? And if we did so, would we still need this duplication check?

Reply:
Removing that code segment and the check here didn't work OOB. Ended up with diffs like below:

<img width="1712" alt="Screen Shot 2023-04-05 at 17 21 26" src="https://user-images.githubusercontent.com/9900707/230110152-b44e4d63-71eb-4407-8cc6-387cc790f097.png">

I need to investigate further and remember where those previous additions are coming from.

Reply:
I added a comment on where those previous additions are coming from.

Reply:
Thank you @turkenh for giving it a try. I believe we can rename those types into common ones once we understand the implications as they are now shared between `spec.forProvider` and `status.atProvider`, when the `status.atProvider` becomes a superset.